kura/org.eclipse.kura.core.cloud/src/main/java/org/eclipse/kura/core/cloud/CloudClientImpl.java

```diff
@@ -124,16 +121,14 @@ public int publish(String appTopic, KuraPayload payload, int qos, boolean retain
     @Override
     public int publish(String deviceId, String appTopic, KuraPayload payload, int qos, boolean retain, int priority)
             throws KuraException {
-        boolean isControl = false;
-        String fullTopic = encodeTopic(deviceId, appTopic, isControl);
         byte[] appPayload = this.cloudServiceImpl.encodePayload(payload);
-        return this.dataService.publish(fullTopic, appPayload, qos, retain, priority);
+        return publish(deviceId, appTopic, appPayload, qos, retain, priority);
     }
 
     @Override
     public int publish(String topic, byte[] payload, int qos, boolean retain, int priority) throws KuraException {
         CloudServiceOptions options = this.cloudServiceImpl.getCloudServiceOptions();
-        return publish(options.getTopicClientIdToken(), payload, qos, retain, priority);
+        return publish(options.getTopicClientIdToken(), topic, payload, qos, retain, priority);
```

Comment:
nice catch!
Can you please review the implementation and rename all ``topic`` parameters to ``appTopic``. Please check consistency with the CloudService Javadoc.